tests/cupy_tests/core_tests/test_ndarray_unary_op.py

```diff
@@ -14,25 +14,25 @@ class TestArrayBoolOp(unittest.TestCase):
     @testing.for_all_dtypes()
     def test_bool_empty(self, dtype):
         with testing.assert_warns(DeprecationWarning):
-            assert not bool(cupy.array((), dtype=dtype))
+            assert bool(cupy.array((), dtype=dtype)) is False
```

Comment:
`bool(...)` is obviously a bool object.

Reply:
Thanks.  You are right.

`ndarray` implements `__nonzero__` method, which was for Python 2.  It may return int, but `bool` calls always return bool.